src/server/main.py

```diff
@@ -33,46 +33,30 @@ def decorated(*args: tuple, **kwargs: dict) -> Response:
 
         # If the token is not passed return message and exit function
         if not token:
-            return make_response(
-                {"message": "Token is missing"},
-                401,
-                {"WWW-Authenticate": 'Basic realm ="Token is missing!"'},
-            )
+            return responses.token_missing()
 
         # If the token is revoked return message and exit function
         if is_token_revoked(token):
-            return make_response(
-                {"message": "Token is revoked"},
-                401,
-                {"WWW-Authenticate": 'Basic realm ="Token is revoked!"'},
-            )
+            return responses.token_revoked()
 
         # Decode the token and retrieve the information contained in it
         try:
             data: dict[str, Any] = jwt.decode(token, app.config["SECRET_KEY"], algorithms=["HS256"])
         except jwt.InvalidTokenError as e:
-            return make_response(
-                {"message": f"Token is invalid: {e}"},
-                401,
-                {"WWW-Authenticate": 'Basic realm ="Token is invalid!"'},
-            )
+            return responses.token_invalid(e)
 
         user_uuid: str = data["uuid"]
 
         with DatabaseProvider.query(QUERIES.SELECT_USER_UUID, (user_uuid,)) as response:
             if response.message is not Message.OK:
-                return make_response({"message": f"Internal error: {response.message}"}, 500)
+                return responses.internal_database_error(response.message)
             user_exists: bool = response.data != []
 
         if not user_exists:
-            return make_response(
-                {"message": "User does not exist"},
-                401,
-                {"WWW-Authenticate": 'Basic realm ="User does not exist!"'},
-            )
+            return responses.user_does_not_exist()
 
         # Returns the current logged-in users context to the routes
-        return fun(user_uuid, *args, **kwargs)
+        return fun(user_uuid, token, *args, **kwargs)
```

Comment:
Why return token?

Reply:
This way functions like `logout` and `refresh` use the token that was checked and provided by the decorator – shorter code in main and smaller chance for a mistake because the token is only retrieved once